lib/fictium/exporters/api_blueprint_exporter.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'api_blueprint_exporter/header_formatter'
+require_relative 'api_blueprint_exporter/resource_formatter'
+require_relative 'api_blueprint_exporter/action_formatter'
+require_relative 'api_blueprint_exporter/example_formatter'
+require_relative 'api_blueprint_exporter/footer_formatter'
+
+module Fictium
+  class ApiBlueprintExporter
+    def export(document)
+      result = process_file(document).presence || ''
+      FileUtils.mkdir_p(File.dirname(export_file))
+      File.write(export_file, result)
+    end
+
+    private
+
+    def process_file(document)
+      [build_header(document), build_footer(document), build_resources(document)]
+        .select(&:present?)
```

Comment:
:eyes: You often are using `select(&:present?).join("\n\n")`, you can delegate it in BaseExporter